includes/admin/extensions/class-extension-manager.php

```diff
@@ -0,0 +1,520 @@
+				'installing'               => __( 'Installing', 'easy-digital-downloads' ),
+				'plugin_install_failed'    => __( 'Could not install the plugin. Please download and install it manually via Plugins > Add New.', 'easy-digital-downloads' ),
+				'extension_install_failed' => sprintf(
+					/* translators: 1. opening anchor tag, do not translate; 2. closing anchor tag, do not translate */
+					__( 'Could not install the extension. Please %1$sdownload it from your account%2$s and install it manually.', 'easy-digital-downloads' ),
+					'<a href="https://easydigitaldownloads.com/your-account/" target="_blank" rel="noopener noreferrer">',
+					'</a>'
+				),
+				'extension_manager_nonce'  => wp_create_nonce( 'edd_extensionmanager' ),
+			)
+		);
+	}
+
+	/**
+	 * Outputs a standard extension card.
+	 *
+	 * @since 2.11.x
+	 * @param object $product             The extension information retrieved from the Products API.
+	 * @param array  $inactive_parameters The array of information to build the button for an inactive/not installed plugin.
+	 * @param array  $active_parameters   The array of information needed to build the link to configure an active plugin.
+	 * @return void
+	 */
+	public function do_extension_card( $product, $inactive_parameters, $active_parameters ) {
+		if ( ! $product ) {
+			return;
+		}
+		$product    = $this->get_card_data( $product );
+		$card_class = 'edd-extension-manager__card';
+		if ( ! empty( $product['card'] ) ) {
+			$card_class .= " {$card_class}--{$product['card']}";
+		}
+		?>
+		<div class="<?php echo esc_attr( $card_class ); ?>">
+			<h3 class="edd-extension-manager__title"><?php echo esc_html( $product['title'] ); ?></h3>
+			<div class="edd-extension-manager__body">
+				<?php if ( ! empty( $product['image'] ) ) : ?>
+					<div class="edd-extension-manager__image">
+						<img alt="" src="<?php echo esc_url( $product['image'] ); ?>" />
+					</div>
+				<?php endif; ?>
+				<div class="edd-extension-manager__content">
+					<?php if ( ! empty( $product['description'] ) ) : ?>
+						<p class="edd-extension-manager__description"><?php echo wp_kses_post( $product['description'] ); ?></p>
+					<?php endif; ?>
+					<div class="edd-extension-manager__group">
+						<?php
+						if ( ! $this->is_plugin_active( $product['basename'] ) ) {
+							?>
+							<div class="edd-extension-manager__step">
+								<?php $this->button( $inactive_parameters ); ?>
+							</div>
+							<?php
+						}
+						?>
+						<div class="edd-extension-manager__step">
+							<?php $this->link( $active_parameters ); ?>
+						</div>
+					</div>
+				</div>
+			</div>
+		</div>
+		<?php
+	}
+
+	/**
+	 * Gets the extension card data.
+	 * Generally, the data from the products API will be used,
+	 * but extending classes can override that in the configuration array.
+	 *
+	 * @since 2.11.x
+	 * @param object $product The extension data from the Products API.
```

Comment:
```suggestion
	 * @param array $product The extension data from the Products API.
```

It's being treated as an array in the method, so I assume it is one.

Reply:
Yes, it is one--for a while it had been an object, but then I decided I didn't want to store all of the API data and since I wanted to potentially merge it with the custom configuration array, I changed it to an array.